Lib/profile/sample.py

```diff
@@ -712,19 +733,59 @@ def main():
 
     sort_value = args.sort if args.sort is not None else 2
 
-    sample(
-        args.pid,
-        sample_interval_usec=args.interval,
-        duration_sec=args.duration,
-        filename=args.outfile,
-        all_threads=args.all_threads,
-        limit=args.limit,
-        sort=sort_value,
-        show_summary=not args.no_summary,
-        output_format=args.format,
-        realtime_stats=args.realtime_stats,
-    )
+    if not(args.pid or args.module or args.script):
+        parser.error(
+            "You must specify either a process ID (-p), a module (-m), or a script to run."
+        )
 
+    if args.pid:
+        sample(
+            args.pid,
+            sample_interval_usec=args.interval,
+            duration_sec=args.duration,
+            filename=args.outfile,
+            all_threads=args.all_threads,
+            limit=args.limit,
+            sort=sort_value,
+            show_summary=not args.no_summary,
+            output_format=args.format,
+            realtime_stats=args.realtime_stats,
+        )
+    elif args.module or args.script:
+        if args.module:
+            cmd = [sys.executable, "-m", *args.module]
+        else:
+            cmd = [sys.executable, *args.script]
+
+        process = subprocess.Popen(cmd)
+
+        try:
+            exit_code = process.wait(timeout=0.1)
```

Comment:
Hummm, why is this? Just in case the process fails immediately?

Reply:
This was to wait until the interpreter in the subprocess has initialised. I replaced it now with a function which retries as a couple of times. Do you think there a better way to handle this?